dali/operators/imgcodec/decoder.cc

```diff
@@ -0,0 +1,411 @@
+to be reallocated to decode the image.
+
+If a value greater than 0 is provided, the operator preallocates two (because of double-buffering)
+host-pinned buffers of the requested size per thread. If selected correctly, no additional
+allocations will occur during the pipeline execution. One way to find the ideal value is to
+do a complete run over the dataset with the ``memory_stats`` argument set to True, and then copy
+the largest allocation value that is printed in the statistics.)code",
+      8*1024*1024)  // based on ImageNet heuristics (8MB)
+  .AddOptionalArg("host_memory_padding_jpeg2k",
+      R"code(Applies **only** to the ``mixed`` backend type.
+
+The padding for nvJPEG2k's host memory allocations, in bytes. This parameter helps to prevent
+the reallocation in nvJPEG2k when a larger image is encountered, and the internal buffer needs
+to be reallocated to decode the image.
+
+If a value greater than 0 is provided, the operator preallocates the necessary number of buffers
+according to the hint provided. If the value is correctly selected, no additional
+allocations will occur during the pipeline execution. One way to find the ideal value is to
+do a complete run over the dataset with the ``memory_stats`` argument set to True, and then copy
+the largest allocation value that is printed in the statistics.)code",
+      0)
+  .AddOptionalArg("hw_decoder_load",
+      R"code(The percentage of the image data to be processed by the HW JPEG decoder.
+
+Applies **only** to the ``mixed`` backend type in NVIDIA Ampere GPU architecture.
+
+Determines the percentage of the workload that will be offloaded to the hardware decoder,
+if available. The optimal workload depends on the number of threads that are provided to
+the DALI pipeline and should be found empirically. More details can be found at
+https://developer.nvidia.com/blog/loading-data-fast-with-dali-and-new-jpeg-decoder-in-a100)code",
+      0.65f)
+  .AddOptionalArg("preallocate_width_hint",
+      R"code(Image width hint.
+
+Applies **only** to the ``mixed`` backend type in NVIDIA Ampere GPU architecture.
+
+The hint is used to preallocate memory for the HW JPEG decoder.)code",
+      0)
+  .AddOptionalArg("preallocate_height_hint",
+      R"code(Image width hint.
+
+Applies **only** to the ``mixed`` backend type in NVIDIA Ampere GPU architecture.
+
+The hint is used to preallocate memory for the HW JPEG decoder.)code",
+      0)
+  .AddOptionalArg("affine",
+      R"code(Applies **only** to the ``mixed`` backend type.
+
+If set to True, each thread in the internal thread pool will be tied to a specific CPU core.
+Otherwise, the threads can be reassigned to any CPU core by the operating system.)code",
+      true)
+  .AddOptionalArg("split_stages",
+      R"code(Applies **only** to the ``mixed`` backend type.
+
+If True, the operator will be split into two sub-stages: a CPU and GPU one.)code",
+      false)
+  .DeprecateArg("split_stages")  // deprecated in DALI 1.0
```

Comment:
Do we want to port deprecated parameters to the experimental decoder?

Reply:
My understanding is that at the moment we want to mimic the behaviour of the old operator, to replace it one day without breaking changes. That's why I copied the whole docs, as I'd expect that somebody (probably not us, as we have two days left) will finish our work and make the new operator behave exactly like the old one